src/lib/Deferred.js

```diff
@@ -1,13 +1,8 @@
-/**
- * Constructor function for Deferred objects.
- *
- * @constructor
- */
-function Deferred () {
+const Deferred = function() {
   const obj = this
-  obj.promise = new Promise(function (resolve, reject) {
+  obj.promise = new Promise(function(resolve, reject) {
     Object.assign(obj, { resolve, reject })
   })
 }
 
-module.exports = Deferred
\ No newline at end of file
+module.exports = Deferred
```

Comment:
not quite sure if this is here for a particular reason, or if we can drop it in favor of plain native Promises

Reply:
Let's try to remove it and use native `Promise`s.